### PR TITLE
chore: add instructions to pip installs and group extras

### PR DESCRIPTION
**Pull request by anna-charlotte**
Goals:

- [x] group extras as follows:
  - [x] We keep default with minimal + protobuf and maybe some others, i.e. `[common]`
  - [x] Minimal real minimal (just `docarray`)
  - [x] One type per modality (`[video]`, `[audio]`, `[text]`, `[image]`)
  - [x] all modalities + some others (`[full]`)
  - [x] One type per DocumentIndex (`[qdrant]`, `[hnswlib]`, `[weaviate]`, `[elasticsearch]`)
  - [x] One for torch, one for tensorflow
  - [x] One type per (push/pull) backend
- [x] Add instruction to error message if a necessary library is not installed (with instructions)
  - For torch, we don't want to throw an error for all failed import, e.g. in `ImageDoc` it would be ok if torch is not imported, we just want to check if it is or not. However, for `TorchTensor` usage torch is needed, and we want to throw an error with instructions.
  
Example for error message with instructions: HF transofrmer
If you install only `transformers ` instead of `transformers[torch]`, and then try to run a minimal example:
```python
from transformers import pipeline

speech_recognizer = pipeline("automatic-speech-recognition", model="facebook/wav2vec2-base-960h")
```
It will crash with:
1. A logger warning:
```python
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
```
2. And a following RuntimeError:
```python
Traceback (most recent call last):
  File "/Users/charlottegerhaher/Desktop/jina-ai/test_mypy_with_docarray_v2/test_docarray.py", line 3, in <module>
    speech_recognizer = pipeline("automatic-speech-recognition", model="facebook/wav2vec2-base-960h")
  File "/Users/charlottegerhaher/Library/Caches/pypoetry/virtualenvs/test-mypy-with-docarray-v2-xw1eJeA4-py3.10/lib/python3.10/site-packages/transformers/pipelines/__init__.py", line 776, in pipeline
    framework, model = infer_framework_load_model(
  File "/Users/charlottegerhaher/Library/Caches/pypoetry/virtualenvs/test-mypy-with-docarray-v2-xw1eJeA4-py3.10/lib/python3.10/site-packages/transformers/pipelines/base.py", line 214, in infer_framework_load_model
    raise RuntimeError(
RuntimeError: At least one of TensorFlow 2.0 or PyTorch should be installed. To install TensorFlow 2.0, read the instructions at https://www.tensorflow.org/install/ To install PyTorch, read the instructions at https://pytorch.org/.

```

- [ ] check and update documentation, if required. See [guide](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md#documentation-guidelines)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -28,20 +28,19 @@ smart-open = {version = ">=6.3.0", extras = ["s3"], optional = true}
 jina-hubble-sdk = {version = ">=0.34.0", optional = true}
 
 [tool.poetry.extras]
-common = ["protobuf", "lz4"]
-torch = ["torch"]
+common = ["protobuf", "lz4", "pandas"]
```
[the remark this answers is not in the document]
Comment (samsja) [reply pyproject.toml:31]: outside of common

**Review by JohannesMessner — APPROVED** [review]

File: tests/documentation/test_docstring.py [hunk 2]
```diff
@@ -47,6 +47,9 @@ def get_obj_to_check(lib):
 for obj in obj_to_check:
     members.extend(get_codeblock_members(obj))
 
+print(f"len(members) = {len(members)}")
```
Comment (samsja) [inline tests/documentation/test_docstring.py:50]: can be removed

File: tests/units/typing/tensor/test_cross_backend.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 try:
     from docarray.typing import TensorFlowTensor
-except (ImportError, TypeError):
+except RuntimeError:
```
Comment (samsja) [inline tests/units/typing/tensor/test_cross_backend.py:9]: why runtime error ?

**Review by samsja — CHANGES_REQUESTED** [review]
added some commenbts

**Review by samsja — APPROVED** [review]

**Pull request merged**